src/shared/components/palette/case-flag/components/select-flag-type/select-flag-type.component.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
+import { SelectFlagTypeErrorMessage } from '../..';
+import { SelectFlagTypeComponent } from './select-flag-type.component';
+
+describe('SelectFlagTypeComponent', () => {
+  let component: SelectFlagTypeComponent;
+  let fixture: ComponentFixture<SelectFlagTypeComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [ReactiveFormsModule],
+      schemas: [CUSTOM_ELEMENTS_SCHEMA],
+      declarations: [SelectFlagTypeComponent]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SelectFlagTypeComponent);
+    component = fixture.componentInstance;
+    component.formGroup = new FormGroup({
+      'flagType': new FormControl(''),
+      'urgent-case': new FormControl(''),
+      'vulnerable-user': new FormControl(''),
+      'other': new FormControl(''),
+      'otherFlagTypeDescription': new FormControl('')
+    });
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should set flag type selected if radio button selected', () => {
+    const radioOtherElement = fixture.debugElement.nativeElement.querySelector('#flag-type-other');
+    radioOtherElement.click();
+    expect(component.flagTypeSelected).toEqual('other');
+  });
+
+  it('should fail validation if flag type is not selected', () => {
+    spyOn(component, 'onNext');
+    const nextButtonElement = fixture.debugElement.nativeElement.querySelector('.button');
+    nextButtonElement.click();
+    expect(component.onNext).toHaveBeenCalled();
+  });
```

Comment:
@johnbenjamin-hmcts - should there be an additional expectation here? This test just seems to check `onNext()` is called but the test description is expecting a validation failure?

Reply:
@Daniel-Lam - this test is no longer required as the validations are tested in other tests. I started with this test initially but then forgot to remove it.